### PR TITLE
Task 5 document exported functions

### PR DESCRIPTION
**Pull request by jpigree**
Remove all warnings showed by GoLint.

### CODE REVIEW

File: agentiface/cli.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package agentiface
 
 import "github.com/spf13/cobra"
 
+// Cli is an interface which encapsulate command line argument parsing.
```
Comment (ccamel) [inline agentiface/cli.go]: encapsulate**s**

File: agentiface/message.go [hunk 2]
```diff
@@ -14,54 +14,69 @@
...
+	// StateConnected represent an agent state when connected.
+	StateConnected State = iota + 1
 
-	EXCHANGE_COMMAND = "crucibuild.command"
-	EXCHANGE_EVENT   = "crucibuild.event"
+	// StateDisconnected represent an agent state when disconnected.
+	StateDisconnected
 
-	MIMETYPE_AVRO = "application/vnd.apache.avro+binary"
+	// ExchangeCommand is the name of the AMQP exchange used by the agent to send commands.j
```
Comment (ccamel) [inline agentiface/message.go]: remove trailing 'j' character

**Comment by jpigree** [comment]
As awaited, it is a very big pull request. However most of the modifications are pedantic. The most important part would be the relevance of the comments.

**Comment by jpigree** [comment]
Okay somehow, I did decrease the coverage (I removed an unused empty fonction). i will just add a unittest to counter this.

**Comment by jpigree** [comment]
Added a unittest to compensate.

File: agentimpl/message.go [hunk 3]
```diff
@@ -158,7 +158,7 @@ func (a *AMQP) Connect() (err error) {
 	endpoint := a.agent.GetConfigString("endpoint")
 
 	if a.State() == agentiface.StateConnected {
-		err = errors.New(fmt.Sprintf("Already connected to: %s", endpoint))
+		err = fmt.Errorf(fmt.Sprintf("Already connected to: %s", endpoint))
```
Comment (ccamel) [inline agentimpl/message.go]: *Errorf()* supports message formating (according to [doc](https://golang.org/pkg/fmt/#Errorf)). So *Sprintf()* is not necessary.

This sentence:
```go
fmt.Errorf("Already connected to: %s", endpoint)
```
should work identically.

**Review by ccamel — APPROVED** [review]
It's a real good piece of work !

I I have just a few small remarks, but nothing crucial for the goal of this issue. We can amend it later.

File: agentimpl/message.go (hunk 3 above)
Comment (jpigree) [reply agentimpl/message.go]: yey. Good catch.

**Comment by jpigree** [comment]
I added two more commits to fix the problems you found. I did not rewrite the branch so it can be clear what was done after the review.

**Pull request merged**